src/LevelDBIntegration.js

```diff
@@ -53,10 +53,124 @@ function connectToLevelDb(options) {
   return db;
 }
 
+/*
+ * Get an instance of an opened leveldb that you can operate with
+ */
+function getOpenedDB(path, ipcEvent) {
+  if (path in dbStore === false) {
+    if (ipcEvent) {
+      ipcEvent.returnValue = {
+        status: 'failed',
+        message: 'Cannot find database.'
+      };
+    }
+    return null;
+  }
+  const db = dbStore[path];
+  if (!db.isOpen()) {
+    if (ipcEvent) {
+      ipcEvent.returnValue = {
+        status: 'failed',
+        message: 'The database is not open.'
+      };
+    }
+    return null;
+  }
+  return db;
+}
+
+/*
+ * List the keys inside the leveldb database.
+ */
+function getKeys(options) {
+  const { path, ipcEvent } = { path: './', ipcEvent: null, ...options };
+  const db = getOpenedDB(path, ipcEvent);
+  if (!db) {
+    return;
+  }
+  const keys = [];
+  db.createKeyStream()
+    .on('data', key => {
+      keys.push(key);
+    })
+    .on('error', function(err) {
+      console.log(err);
+      if (ipcEvent) {
+        ipcEvent.returnValue = {
+          status: 'failed',
+          message: 'Cannot fetch keys from the database.'
+        };
+      }
+    })
+    .on('end', function() {
+      if (ipcEvent) {
+        ipcEvent.returnValue = {
+          status: 'success',
+          keys
+        };
+      } else {
+        console.log(`Keys in ${path}: ${keys}`);
+      }
+    });
+}
+
+/*
+ * Get the value given the key
+ */
+function getValue(options) {
+  const { path, key, ipcEvent } = { path: './', key: '', ipcEvent: null, ...options };
+  const db = getOpenedDB(path, ipcEvent);
+  if (!db) {
+    return;
+  }
+  db.get(key, function(err, value) {
+    if (err) {
+      if (err.notFound) {
+        if (ipcEvent) {
+          ipcEvent.returnValue = {
+            status: 'failed',
+            message: `${key} is not in ${path}`
+          };
+        }
+        return;
+      }
+      if (ipcEvent) {
+        ipcEvent.returnValue = {
+          status: 'failed',
+          message: `Failed to get ${key} in ${path}`
+        };
+      }
+      console.log(err);
+      return;
+    }
+
+    if (ipcEvent) {
+      ipcEvent.returnValue = {
+        status: 'success',
+        value
+      };
+    } else {
+      console.log(`${path} -> ${key}: ${value}`);
+    }
+  });
+}
+
 function enableDBConnect() {
   ipcMain.on('connect-to-leveldb', (event, params) => {
     connectToLevelDb({ ...params, ipcEvent: event });
   });
 }
 
-export { enableDBConnect };
+function enableGetKeys() {
+  ipcMain.on('get-db-keys', (event, params) => {
+    getKeys({ ...params, ipcEvent: event });
+  });
+}
+
+function enableGetValue() {
+  ipcMain.on('get-value-by-key', (event, params) => {
+    getValue({ ...params, ipcEvent: event });
+  });
+}
```

Comment:
@emanlodovice We can merge this into just one IPC listener. We can have only one event e.g `leveldb-command` with the following event params:

```js
{
  command: 'connect',  // or get-keys, get-value, etc.
  params: {
    path: '...'
  }
}
```

We can do a switch-case or whatever in the event handler. That way we don't need to call a lot of functions in the main process.

Reply:
Is this a convention @arnellebalane ?

Reply:
@emanlodovice I think it's more of a design pattern. _Command Pattern_? @jjlumagbas ? 😅

Reply:
updated @arnellebalane